### PR TITLE
ref(views): move alert, acceptOrgInvite and issue rule views to tsx

### PR DESCRIPTION
**Pull request by JonasBa**

### CODE REVIEW

File: tests/js/setup.ts
Comment (scttcper) [inline tests/js/setup.ts:195]: i think you can pass params to the router

File: tests/js/setup.ts
Comment (scttcper) [inline tests/js/setup.ts:203]: and then this should be unnecessary

File: tests/js/setup.ts
Comment (scttcper) [inline tests/js/setup.ts:197]: location should be available on router.location

File: tests/js/setup.ts
Comment (scttcper) [inline tests/js/setup.ts:193]: would be kinda nice to use a generic here since which was my thinking here https://github.com/getsentry/sentry/blob/master/tests/js/sentry-test/initializeOrg.tsx#L35 since you get type checking pretty easily

File: tests/js/setup.ts
Comment (scttcper) [inline tests/js/setup.ts:192]: overall in favor of this since `initializeOrg` is stupid

File: static/app/views/alerts/rules/metric/create.spec.tsx
Comment (scttcper) [inline static/app/views/alerts/rules/metric/create.spec.tsx:67]: i think this is the event we use when creating https://github.com/getsentry/sentry/blob/master/static/app/views/alerts/create.tsx#L109
```suggestion
          EventView.fromLocation(TestStubs.routeComponentProps().location)
```

File: static/app/views/alerts/list/rules/index.spec.tsx
Comment (scttcper) [inline static/app/views/alerts/list/rules/index.spec.tsx:91]: i think you want clear
```suggestion
    jest.clearAllMocks();
```

File: tests/js/setup.ts
Comment (JonasBa) [reply tests/js/setup.ts:192]: Yeah, initializeOrg is doing waaay too much

File: tests/js/setup.ts
Comment (JonasBa) [reply tests/js/setup.ts:193]: Hmm, that causes a typeerr on params, I see you had to cast that in your change. Since that exists already I'll do the cast too

File: static/app/views/alerts/list/rules/index.spec.tsx
Comment (JonasBa) [reply static/app/views/alerts/list/rules/index.spec.tsx:91]: Idk, the tests still pass lol

Comment (JonasBa) [reply static/app/views/alerts/list/rules/index.spec.tsx:91]: but yes, clear is more accurate

File: tests/js/setup.ts
Comment (evanpurkhiser) [inline tests/js/setup.ts:192]: Is this different from what comes back from the `initializeOrg` thing?

Comment (JonasBa) [reply tests/js/setup.ts:192]: No, pretty much same thing, but there are tests that don't call initializeOrg and calling TestStubs.routeComponentProps() is more convenient than being forced to create org

**Comment by JonasBa** [comment]
@scttcper mind giving this another look :)?

**Review by scttcper — APPROVED** [review]

**Comment by priscilawebdev** [comment]
this is a lot to review 🙂

**Comment by JonasBa** [comment]
@priscilawebdev it contains 4 stacked pull requests that were merged into this one after all of them were approved :) This way I can merge to master from a single commit and not wait for CI 4 times :)

**Pull request merged**
